### PR TITLE
fix(discover): Fix timepicker bug in Discover

### PR DESCRIPTION
**Pull request by lynnagara**
Prevents Discover crashing for superusers due to an undefined utc
property. This was introduced in
https://github.com/getsentry/sentry/pull/11936.

Fixes JAVASCRIPT-57X

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationDiscover/utils.jsx
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/utils.jsx]: I think this was needed to keep `utc` in the query string across discover queries

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/utils.jsx]: `utc` isn't a valid value in a Discover querystring and would just be ignored anyway

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDiscover/utils.jsx]: I think the date picker needs it though - I will take a look and remove `statsPeriod` and `utc` being required.

**Review by billyvg — APPROVED** [review]
I believe this breaks an edge case in date picker -- if this only happens for superusers we shouldn't merge this yet, otherwise, if the scope is larger than that we may want to merge it.

**Comment by lynnagara** [comment]
@billyvg What's the datepicker edge case?

**Comment by billyvg** [comment]
Go to discover, toggling utc, update, re-open date selector

**Comment by lynnagara** [comment]
It's consistently reproducible for superusers, but I've seen it happen in some other more nuanced cases (e.g. coming from specific pages)

**Comment by lynnagara** [comment]
It's kind of broken anyway since the datepicker doesn't seem to respect `hasCustomRouting` at all and just always updates a URL

**Comment by lynnagara** [comment]
Closed in favor of https://github.com/getsentry/sentry/pull/12102

**Pull request closed without merging**
